backends/core/models.py

```diff
@@ -163,10 +219,11 @@ def _finish(self):
     jobs = jobs.filter(Job.pipeline_id == self.id)
     jobs = jobs.filter(StartCondition.preceding_job_id == None)
     jobs = jobs.options(load_only('status')).all()
-    status = 'succeeded'
+    status = Pipeline.STATUS.SUCCEEDED
     for job in jobs:
-      if job.status == 'failed':
-        status = 'failed'
+      # IDLE means the job has not run at all or it has been cancelled
+      if job.get_status() in [Job.STATUS.FAILED, Job.STATUS.IDLE]:
```

Comment:
If there is a job that did not start at all and the pipeline has been stopped (say due to start conditions unfulfilled), then the pipeline status will be `SUCCEEDED` since there are no `FAILED` jobs with no dependencies, but there are `IDLE` ones. In this case, the pipeline should be `FAILED` because not all the jobs even got the chance to be run.
What do you think? @dulacp

Reply:
[This test would fail](https://github.com/google/crmint/blob/30fcc5569f7022250370e126ba88b1fb5f6025fd/backends/tests/integration/core/models_tests.py#L772)

Reply:
You got a fair point, but this should never happen right?
Because if start conditions are unfulfilled the pipeline will immediately go to failure? either in [worker_failed](https://github.com/google/crmint/pull/19/files#diff-653fbbc4974716642ebfe4e727fb5b60R559) or [worker_succeeded](https://github.com/google/crmint/pull/19/files#diff-653fbbc4974716642ebfe4e727fb5b60R530)?

Is [your test](https://github.com/google/crmint/blob/30fcc5569f7022250370e126ba88b1fb5f6025fd/backends/tests/integration/core/models_tests.py#L772) failing right now?

Reply:
Yes, I have changed the status of a stopping job to FAILED as a fix